ion-schema/src/system.rs

```diff
@@ -987,10 +1122,49 @@ impl SchemaSystem {
     }
 
     /// Requests each of the provided [`DocumentAuthority`]s, in order, to get ISL model for the
-    /// requested schema id until one successfully resolves it.
+    /// requested schema id until one successfully resolves it using ISL 1.0.
+    /// If an authority throws an exception, resolution silently proceeds to the next authority.
+    /// If the given ISL model has any ISL 2.0 related types/constraints, resolution returns an error.
+    pub fn load_isl_schema_v1_0<A: AsRef<str>>(&mut self, id: A) -> IonSchemaResult<IslSchema> {
```

Comment:
I'm a little confused by the function signature here. Does this mean that you need to know the ISL version of e.g. `foo.isl` before loading `foo.isl` so that you call the correct function? If so, we need to change it—it's not reasonable for a user to have to determine the ISL version _before_ loading the schema.

Reply:
Oh yes! Thanks for catching this. I can probably add the check inside `load_isl_schema` for version marker one doesn't need to know what version it is using.